### ISSUE TITLE
Support annotations, labels and tls elements for k8s security groups (ingress resources)

### ISSUE THREAD

**Issue by mdirkse** [spinnaker/spinnaker#2000]
We're using Spinnaker to deploy to GKE. The security groups that are created by Spinnaker work well for HTTP traffic, but we want to support HTTPS traffic as well. In our case (we're using the [Nginx ingress controller](https://github.com/kubernetes/ingress/tree/master/controllers/nginx) and automatic certificate generation), so to support TLS we'll need to be able to configure the `annotations`, `labels` and `tls` elements. The extra elements would look something like this:

```yaml
annotations:
  kubernetes.io/ingress.class: "internal"

labels:
  stable.k8s.psg.io/kcm.class: default

tls:
- hosts:
  - app.localdomain
  secretName: tsl-app-localdomain
```

I think it would make the most sense for the annotations and labels to be separate headings in the security group edit screen, and then have a "Secret name" field per rule where the user can enter the name of a secret. A "?" button should be added to explain that Spinnaker will not create the secret for you.

Taking a crack at this myself (with a little help from @ethanfrogers). I believe this will also solve #1338.

**Comment by mdirkse** [comment]
![screenshot from 2017-09-30 16-47-00](https://user-images.githubusercontent.com/312638/31046765-062aee5c-a5ff-11e7-857c-8f404ce10b82.png)

Mock up of the UI

**Comment by ethanfrogers** [comment]
This looks awesome @mdirkse! I'd put Labels and Annotations under an "Advanced Settings" section to stay consistent with other areas in the UI. I can't wait to see the final product!

**Comment by lwander** [comment]
:+1: nice work!

**Comment by mdirkse** [comment]
Updated UI.

![screenshot from 2017-10-09 16-05-40](https://user-images.githubusercontent.com/312638/31342029-bff5257e-ad0b-11e7-99e5-50f58729fa3c.png)

**Comment by mdirkse** [comment]
@lwander out of curiosity, what is this for?

https://github.com/spinnaker/clouddriver/blob/7fa8affbd6cb6e12b6ec0133fd6ad3b7fd79e5d6/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/description/securitygroup/KubernetesSecurityGroupDescription.groovy#L46

**Comment by lwander** [comment]
If it doesn't match anything in the k8s spec, it's likely because it got deprecated. This was initially implemented around k8s 1.1/1.2

**Comment by mdirkse** [comment]
Final version of UI

![screenshot from 2017-10-20 13-36-08](https://user-images.githubusercontent.com/312638/31820029-972014e6-b59f-11e7-9f84-f1a3ac8cfa7c.png)
![screenshot from 2017-10-20 13-38-03](https://user-images.githubusercontent.com/312638/31820030-973897f0-b59f-11e7-8b77-8a187700570b.png)
